.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,118 @@
+name: Build and Test
+
+on:
+  workflow_call:
+    inputs:
+      sha:
+        required: true
+        type: string
+      ENVIRONMENT: 
+        required: true
+        type: string
+    secrets:
+      API_BASE_URL:
+        required: true
+      STORY_PROTOCOL_CONTRACT:
+        required: true
+      IP_ASSET_REGISTRY_CONTRACT:
+        required: true
+      IP_ORG_CONTROLLER_CONTRACT:
+        required: true
+      RELATIONSHIP_MODULE_CONTRACT:
+        required: true
+      REGISTRATION_MODULE_CONTRACT:
+        required: true
+      LICENSE_REGISTRY_CONTRACT:
+        required: true
+      MODULE_REGISTRY_CONTRACT:
+        required: true
+      LICENSE_MODULE_CONTRACT:
+        required: true
+      RPC_PROVIDER_URL:
+        required: true
+      WALLET_PRIVATE_KEY:
+        required: true
+      TEST_WALLET_ADDRESS:
+        required: true 
+      TEST_IPORG_ID:
+        required: true 
+      TEST_IPORG_ID_WITH_HOOK:
+        required: true
+      TEST_IPASSET_ID1:
+        required: true
+      TEST_IPASSET_ID2:
+        required: true
+      TEST_RELATIONSHIP_ID:
+        required: true
+      TEST_RELATIONSHIP_TYPE:
+        required: true
+      TEST_LICENSE_ID:
+        required: true
+      TEST_HOOK_ID:
+        required: true
+      TEST_MODULE_ID:
+        required: true
+      TEST_TRANSACTION_ID:
+        required: true
+
+jobs:
+  build:
+    name: Build and Test
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    ## Example to fix envrionment secret not passing in: https://github.com/AllanOricil/workflow-template-bug/blob/master/.github/workflows/workflow-template-fix-without-required-secret.yml
+    environment: ${{ inputs.ENVIRONMENT }}
+    env:
+      ## To use Remote Caching, uncomment the next lines and follow the steps below.
+      # TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
+      # TURBO_TEAM: ${{ vars.TURBO_TEAM }}
```

Comment:
are these two above meant to be commented out?

Reply:
These were added by Bruce. Yeah we can remove them later @AndyBoWu